sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -641,6 +661,18 @@ static struct snd_soc_dai_link *sof_card_dai_links_create(struct device *dev,
 	links[id].no_pcm = 1;
 	links[id].cpus = &cpus[id];
 	links[id].num_cpus = 1;
+
+	port = sof_rt5682_check_ssp_port_overwrite(dev, links[id].codecs);
+	if (port >= 0) {
+		dev_info(dev, "overwrite codec ssp port %d\n", port);
+		ssp_codec = port;
+	}
+
+	links[id].name = devm_kasprintf(dev, GFP_KERNEL,
+					"SSP%d-Codec", ssp_codec);
```

Comment:
Question: for this machine you would still need a topology that hard-codes the same value as the _DSD property, yes?

In other words, CoreBoot and topology have to match now, and the machine driver would be generic?

Reply:
yes, we should be able to remove most of dmi quirks due to different ssp port assignment.